backend-tests/tests/test_deployments.py

```diff
@@ -0,0 +1,513 @@
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=2)).isoformat("T")) + "Z",
+                        # Batch size is optional in the last stage (ie, it is the remaining devices)
+                    },
+                ],
+            },
+            {
+                "name": "Three phases, first batch, last start",
+                "artifact_name": "deployments-phase-testing",
+                "device_count": 100,
+                "phases": [
+                    {
+                        "batch_size": 10,
+                        # The nr of devices that asked for an update within the phase, in this case 0
+                        "device_count": 0,
+                    },
+                    {
+                        "batch_size": 45,
+                        "device_count": 0,
+                    },
+                    {
+                        "batch_size": 45,
+                        "device_count": 0,
+                    }
+                ],
+            }),
+        # Phase, Five batches, just make sure it works. Should behave like all other > 1 cases
+        (
+            {
+                "name":
+                "Five phases, first no start time, last start time, no batch size",
+                "artifact_name":
+                "deployments-phase-testing",
+                "devices": ["dummyuuid" + str(i) for i in range(100)],
+                "phases": [
+                    {
+                        # Start time is optional in the first stage (default to now)
+                        "batch_size": 10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=2)).isoformat("T")) + "Z",
+                        # Batch size is optional in the last stage (ie, it is the remaining devices)
+                    },
+                ],
+            },
+            {
+                "name":
+                "Five phases, first no start time, last start time, no batch size",
```

Comment:
formatting again - pep or not, this is dumb and confusing, please move to 1 line